app/models/pipeline_run_stage.rb

```diff
@@ -168,37 +162,15 @@ def update_job_status
     if failed? || succeeded?
       unless job_log_id
         # set log id if not set
-        _job_status, self.job_log_id, _job_hash, self.job_description = job_info(job_id, id)
+        _job_status, self.job_log_id, self.job_description = job_info(job_id, id)
         save
       end
       return
     end
     # The job appears to be in progress.  Check to make sure it hasn't been killed in AWS.   But not too frequently.
     return unless due_for_aegea_check?
-    self.job_status, self.job_log_id, job_hash, self.job_description = job_info(job_id, id)
-    if [STATUS_ERROR, STATUS_FAILED].include?(job_status)
-      save
-      return
-    end
-    unless instance_terminated?(job_hash)
-      save
-      return
-    end
-
-    # This code path is not being used anymore because we no longer use spot
-    # instances for the Batch jobs. Spot instances can be terminated at any time
-    # if someone bids more, so a retry is necessary. Now we use on-demand
-    # instances. If we decided to go back to spot instances, this code would
-    # ensure that terminated Batch jobs get resubmitted.
-
-    # note failed attempt and retry
-    add_failed_job
-    unless count_failed_tries <= MAX_RETRIES
-      LogUtil.log_err_and_airbrake("Job #{job_id} for pipeline run #{id} was killed #{MAX_RETRIES} times.")
-      save
-      return
-    end
-    run_job # this saves
+    self.job_status, self.job_log_id, self.job_description = job_info(job_id, id)
```

Comment:
How can we be sure that the code below that triggered run_job 3x times would not happen ?
If it did, do we get an alert to restart the job?

Reply:
That code would only run when the ec2 instance was terminated, and we already have auto-restarts in a different part of our code. So this is redundant.